titiler/pgstac/factory.py

```diff
@@ -31,15 +30,16 @@ def PathParams(searchid: str = Path(..., description="Search Id")) -> str:
     return searchid
 
 
-def SearchParams(body: SearchQuery) -> Tuple[str, Optional[Dict[str, Any]]]:
+def SearchParams(
+    body: model.RegisterMosaic,
+) -> Tuple[model.PgSTACSearch, model.Metadata]:
     """Search parameters."""
-    search = body.json(
+    search = body.dict(
         exclude_none=True,
         exclude={"metadata"},
         by_alias=True,
     )
-    metadata = body.metadata or {}
-    return search, metadata
+    return model.PgSTACSearch(**search), body.metadata
```

Comment:
FYI: @drnextgis

Reply:
Awesome! As soon as it is merged, I will update the example in the documentation.

Reply:
https://github.com/stac-utils/titiler-pgstac/commit/d3eecdcbcf23aef59fd240914441ef27b3e885ee#diff-0f0c40de44a0841ed5180f16d66212d1cece2a7d4400789007c2f73bf8432622L39 I updated the docs 🙏